### DISCUSSION TITLE
Why use filterHTMLPlugin to remove all html

### DISCUSSION THREAD

**Discussion by z-950** [Milkdown/milkdown#644] [General]
Markdown support html tags.

But when i paste markdown with some tags in Milkdown, the tags are removed.

And i checked the [Remark ast result](https://astexplorer.net/#/gist/0a92bbf654aca4fdfb3f139254cf0bad/9169a791037b71bfcfeec55d363a5c8c2adf988a). Remark can parser html tag.

So i wonder why you remove html, thanks.

And then i think i may can make a plugin to enhance it.

**Comment by Saul-Mirone** [comment]
I remove html for several reasons:

1. It would be easier for xss attack.
2. Since html can be mixed with markdown, the user experience would be very hard to design.

However, I think we can resolve them now. For issue 1, I think we can provide users an option to open it. And we'll let the users know that enable html may cause some dangerous. For issue 2, I still don't have a really good solution.

If you have interests in implement this. Maybe you can have a try.

**Reply by quank123wip** [reply]
Using some filter can reduce xss attack and keep some specific nodes? I'm not pretty sure due new xss exploits are being mined everyday. But exactly we can provide plugins to replace most html markups.

**Reply by rabble-coop** [reply]
> For issue 1, I think we can provide users an option to open it. And we'll let the users know that enable html may cause some dangerous.

This would be great, I really need to be able to control the sanitisation. Is there any workaround for this in the meantime?

**Reply by Saul-Mirone** [reply]
> > For issue 1, I think we can provide users an option to open it. And we'll let the users know that enable html may cause some dangerous.
> 
> This would be great, I really need to be able to control the sanitisation. Is there any workaround for this in the meantime?

There isn't right now. The good news is that I'm already working on the new version which will support HTML.

**Reply by rumack** [reply]
Great news, looking forward to that. I'm enjoying using Milkdown.
